include/seqan3/alphabet/aminoacid/aa20.hpp

```diff
@@ -0,0 +1,425 @@
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \author Joshua Kim <joshua.kim AT fu-berlin.de>
+ * \brief Contains seqan3::aa20, container aliases and string literals.
+ */
+
+#pragma once
+
+#include <cassert>
+#include <vector>
+
+#include <seqan3/core/platform.hpp>
+#include <seqan3/alphabet/detail/convert.hpp>
+#include <seqan3/alphabet/aminoacid/concept.hpp>
+
+namespace seqan3
+{
+/*!\brief The canonical amino acid alphabet
+ * \ingroup aminoacid
+ * \implements seqan3::aminoacid_concept
+ *
+ * \details
+ * The alphabet consists of letters A, C, D, E, F, G, H, I, K, L, M, N, P, Q, R, S, T, V, W, Y
+ *
+ * The alphabet may be brace initialized from the static letter members (see above). Note that you cannot assign
+ * regular characters, but additional functions for this are available.
+ *
+ *```cpp
+ *     aa20 my_letter{aa20::A};
+ *     // doesn't work:
+ *     // aa20 my_letter{'A'};
+ *
+ *     my_letter.assign_char('C'); // <- this does!
+ *
+ *     my_letter.assign_char('?'); // converted to A internally
+ *     if (my_letter.to_char() == 'A')
+ *        std::cout << "yeah\n"; // "yeah";
+ *```
+ *
+ * <b>Note</b>: Letters not belonging to the canonical alphabet will be automatically translated. Letters
+ * which belong in the extended alphabet will be automatically converted based on the frequency of their options.<br>
+ * Terminator characters are converted to W, because the most commonly occurring
+ * stop codon in higher eukaryotes is UGA<sup>2</sup>.
+ * Anything unknown is converted to S, because it occurs most frequently across 53 vertebrates<sup>1</sup>.
+ *
+ * <table>
+ * <tr><th>Input Letter     <th>Converts to
+ * <tr><td>B                <td>D<sup>1</sup>
+ * <tr><td>J                <td>L<sup>1</sup>
+ * <tr><td>O                <td>L<sup>1</sup>
+ * <tr><td>U                <td>C<sup>1</sup>
+ * <tr><td>Z                <td>E<sup>1</sup>
+ * <tr><td>X (Unknown)      <td>S<sup>1</sup>
+ * <tr><td>* (Terminator)   <td>W<sup>2</sup>
+ * </table>
+ * <b><sup>1</sup></b>King, J. L., & Jukes, T. H. (1969). Non-Darwinian Evolution. <br>
+ * Science, 164(3881), 788-798. doi:10.1126/science.164.3881.788<br>
+ * <b><sup>2</sup></b>Trotta, E. (2016). Selective forces and mutational biases drive stop codon usage <br>
+ * in the human genome: a comparison with sense codon usage. <br>
+ * BMC Genomics, 17, 366. http://doi.org/10.1186/s12864-016-2692-4
+ */
```

Comment:
Although inline HTML works fine mostly, we prefer to use Markdown when possible, so `*foo*` instead of `<b>foo</b>` a.s.o. It should also work for tables, but I don't think there is markdown for sup so you can use `<sub>`.

Reply:
Other than that a very good description :+1: